addons/membership/membership.py

```diff
@@ -488,6 +488,15 @@ def action_cancel(self, cr, uid, ids, context=None):
             member_line_obj.write(cr, uid, mlines, {'date_cancel': today})
         return super(Invoice, self).action_cancel(cr, uid, ids, context=context)
 
+    def unlink(self, cr, uid, ids, context=None):
+        member_line_obj = self.pool.get('membership.membership_line')
+        for invoice in self.browse(cr, uid, ids, context=context):
+            mlines = member_line_obj.search(cr, uid,
```

Comment:
Why don't you refactorize the code in one method and rename the variable? That kind of changes are allowed of course in upstream.